packages/server/src/db/utils.ts

```diff
@@ -1,90 +1,64 @@
-const newid = require("./newid")
-const {
-  DocumentType: CoreDocType,
-  InternalTable,
-  getRoleParams,
-  generateRoleID,
-  APP_DEV_PREFIX,
-  APP_PREFIX,
-  SEPARATOR,
-  StaticDatabases,
-  isDevAppID,
-  isProdAppID,
-  getDevelopmentAppID,
-  generateAppID,
-  getQueryIndex,
-  ViewName,
-  getDocParams,
-  getRowParams,
-  generateRowID,
-  getUserMetadataParams,
-  generateUserMetadataID,
-  getGlobalIDFromUserMetadataID,
-} = require("@budibase/backend-core/db")
-
-const UNICODE_MAX = "\ufff0"
-
-const AppStatus = {
+import newid from "./newid"
+import { db as dbCore } from "@budibase/backend-core"
+
+type Optional = string | null
+
+export const AppStatus = {
   DEV: "development",
   ALL: "all",
   DEPLOYED: "published",
 }
 
-const DocumentType = CoreDocType
-
-const SearchIndexes = {
+export const SearchIndexes = {
   ROWS: "rows",
 }
 
-exports.StaticDatabases = StaticDatabases
-
-const BudibaseInternalDB = {
+export const BudibaseInternalDB = {
   _id: "bb_internal",
   type: "budibase",
   name: "Budibase DB",
   source: "BUDIBASE",
   config: {},
 }
 
-exports.APP_PREFIX = APP_PREFIX
-exports.APP_DEV_PREFIX = APP_DEV_PREFIX
-exports.isDevAppID = isDevAppID
-exports.isProdAppID = isProdAppID
-exports.USER_METDATA_PREFIX = `${DocumentType.ROW}${SEPARATOR}${InternalTable.USER_METADATA}${SEPARATOR}`
-exports.LINK_USER_METADATA_PREFIX = `${DocumentType.LINK}${SEPARATOR}${InternalTable.USER_METADATA}${SEPARATOR}`
-exports.TABLE_ROW_PREFIX = `${DocumentType.ROW}${SEPARATOR}${DocumentType.TABLE}`
-exports.ViewName = ViewName
-exports.InternalTables = InternalTable
-exports.DocumentType = DocumentType
-exports.SEPARATOR = SEPARATOR
-exports.UNICODE_MAX = UNICODE_MAX
-exports.SearchIndexes = SearchIndexes
-exports.AppStatus = AppStatus
-exports.BudibaseInternalDB = BudibaseInternalDB
-exports.generateAppID = generateAppID
-exports.generateDevAppID = getDevelopmentAppID
-exports.generateRoleID = generateRoleID
-exports.getRoleParams = getRoleParams
-exports.getQueryIndex = getQueryIndex
-exports.getDocParams = getDocParams
-exports.getRowParams = getRowParams
-exports.generateRowID = generateRowID
-exports.getUserMetadataParams = getUserMetadataParams
-exports.generateUserMetadataID = generateUserMetadataID
-exports.getGlobalIDFromUserMetadataID = getGlobalIDFromUserMetadataID
+export const SEPARATOR = dbCore.SEPARATOR
```

Comment:
The mix of constants and utils in this block can be pretty confusing, especially when you're writing a new constant and deciding where to put it. It's not very concrete if it should go into the service/constants, this utils file, or the backend core. 

Maybe we should deprecate this block, in favour of the core version directly if it exists. Ideally we could namespace out constants in core, like `Constants.DB.X`

Reply:
Yeah I think ideally I would like to move all of this into the backend-core, its a WIP, all of this is just exposed here for now rather than taking on the massive task of getting everything properly exposed from backend-core!